### PR TITLE
Fix crash in tf.nn.atrous_conv2d with large rate

### LINKED ISSUE #46915: tf.nn.atrous_conv2d crashes(aborts) when rate is large value
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below):2.1.0
- Python version:3.7.6
- Bazel version (if compiling from source):N/A
- GCC/Compiler version (if compiling from source):N/A
- CUDA/cuDNN version:N/A
- GPU model and memory:N/A

**Describe the current behavior**
`tf.nn.atrous_conv2d` crashes(aborts) when `rate` is large value

**Describe the expected behavior**
expect an exception message if the input unexpected instead of crash. 

**Standalone code to reproduce the issue**
~~~python
import tensorflow as tf
import numpy as np
tf.nn.atrous_conv2d(value=np.ones((1,1,1,5)), filters=np.ones((1,1,5,1)), rate=2147483647, padding='SAME')
~~~

~~~python
2021-02-04 04:47:25.891213: F tensorflow/core/framework/tensor_shape.cc:353] Check failed: 0 <= new_num_elements (0 vs. -1)
Aborted (core dumped)
~~~

### PR DESCRIPTION
**Pull request by yongtang**
This PR tries to address the issue raised in #46915 where
tf.nn.atrous_conv2d will crash when rate is larger than 2^31.

This PR fixes #46915.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: tensorflow/python/kernel_tests/nn_ops/atrous_conv2d_test.py [hunk 1]
```diff
@@ -196,6 +208,18 @@ def testAtrousConv2DTransposeForward(self):
                     x, f_up, y_shape, strides=[1, 1, 1, 1], padding=padding)
                 self.assertAllClose(y1, y2, rtol=1e-3, atol=1e-3)
 
+  @test_util.run_deprecated_v1
+  def testAtrousConv2DTransposeInvalid(self):
+    with self.session():
```
Comment (cantonios) [inline tensorflow/python/kernel_tests/nn_ops/atrous_conv2d_test.py:213]: Does this needed the `run_deprecated_v1` and `with self.session()`?  We're trying to slowly update all these tests to run in V2 mode.

**Review by cantonios — CHANGES_REQUESTED** [review]

**Comment by yongtang** [comment]
Thanks @cantonios for the review. The `@test_util.run_deprecated_v1` has been removed. Please let me know if there are any other issues needs to be addressed.

**Review by cantonios — APPROVED** [review]
